crates/broker/src/order_picker.rs

```diff
@@ -97,6 +98,7 @@ pub struct OrderPicker<P> {
     priced_orders_tx: mpsc::Sender<Box<OrderRequest>>,
     stake_token_decimals: u8,
     order_cache: OrderCache,
+    active_tasks: Arc<Mutex<HashMap<String, Box<OrderRequest>>>>,
```

Comment:
I don't think you need the mutex logic to track these, as you can just remove the entry when the task is joined. I can make that change in a follow up as I'm changing related things